app/models/logstash_queries/date_range_top_n_field_query.rb

```diff
@@ -1,15 +1,20 @@
 class DateRangeTopNFieldQuery < DateRangeTopNQuery
+  attr_reader :filters
+
   def initialize(affiliate_name, start_date, end_date, filter_field, filter_value, agg_options = {})
     super(affiliate_name, start_date, end_date, agg_options)
-    @filter_field, @filter_value = filter_field, filter_value
+    @filters = { 'affiliate' => @affiliate_name, filter_field => filter_value }.compact
   end
```

Comment:
How about using symbols has hash keys?  { 'name': value, another_name: value } format?

Reply:
I'm using string keys because they need to be strings when passed to the JSON methods:
```ruby
    filters.each do |field, value|
      json.must do
        json.child! { json.term { json.set! field, value } }
      end
    end
```
I could use symbols, but I'd need to call `.to_s` on them to pass them to the JSON.

Reply:
Usually I prefer using the 1.9 and greater ruby syntax and its also faster. But if you feel strongly about it, you can leave it.

Reply:
Actually so does rubocop :) . 
https://github.com/rubocop-hq/ruby-style-guide#symbols-as-keys

Reply:
Yeah, normally I use symbols as keys, but in this case I'd prefer to keep it consistent with the string keys used throughout the analytics JSON, and avoid the extra `.to_s` call.